handler/handler.go

```diff
@@ -39,78 +33,133 @@ func NewBatchHandler(esClient *dpelasticsearch.Client) *BatchHandler {
-				log.Info(ctx, "No title for inbound event, no transformation possible")
-				continue // break here
-			}
+	jsonCreateResponse, _, err := batchHandler.esClient.BulkUpdate(ctx, esDestIndex, esDestURL, bulkcreate)
+	if err != nil {
+		log.Error(ctx, "error in response from elasticsearch while creating the event", err)
+		return err
+	}
 
-			esmodel := t.TransformEventModelToEsModel(events[i])
+	var bulkCreateRes models.EsBulkResponse
+	if err := json.Unmarshal(jsonCreateResponse, &bulkCreateRes); err != nil {
+		log.Error(ctx, "error unmarshaling jsonCreateResponse", err)
+		return err
+	}
 
-			if esmodel != nil {
-				b, err := json.Marshal(esmodel)
+	if bulkCreateRes.Errors {
+		var bulkupdate []byte
+		for _, resCreateItem := range bulkCreateRes.Items {
+			if resCreateItem["create"].Status == 409 {
+				event := documentList[resCreateItem["create"].ID]
+				updateBulkRequestBody, err := prepareEventForBulkRequestBody(ctx, &event, "update")
 				if err != nil {
-					log.Fatal(ctx, "error marshal to json", err)
-					return
+					log.Error(ctx, "error in preparing the bulk for update", err)
+					continue
 				}
-
-				bulk = append(bulk, []byte("{ \"create\": { \"_id\": \""+esmodel.Title+"\" } }\n")...)
-				bulk = append(bulk, b...)
-				bulk = append(bulk, []byte("\n")...)
+				bulkupdate = append(bulkupdate, updateBulkRequestBody...)
+			} else if resCreateItem["create"].Status == 201 {
+				continue
+			} else {
+				log.Error(ctx, "error creating doc to ES", err,
+					log.Data{
+						"response status": resCreateItem["create"].Status,
+						"response.title:": resCreateItem["create"].ID,
+					})
+				continue
 			}
-			i++
 		}
 
-		//@TODO  decision on handle failed updates to elasticsearch later
-		_, _, err := bh.esClient.BulkUpdate(ctx, esDestIndex, esDestURL, bulk)
+		jsonUpdateResponse, _, err := batchHandler.esClient.BulkUpdate(ctx, esDestIndex, esDestURL, bulkupdate)
 		if err != nil {
-			log.Error(ctx, "error in response from elasticsearch", err)
-			return
+			log.Error(ctx, "error in response from elasticsearch while updating the event", err)
+			return err
 		}
 
-		log.Info(ctx, "go routine for inserting into ES ends with insertChannel")
-	}()
+		var bulkUpdateRes models.EsBulkResponse
+		if err := json.Unmarshal(jsonUpdateResponse, &bulkUpdateRes); err != nil {
+			log.Error(ctx, "error unmarshaling bulkUpdateRes", err)
+			return err
+		}
 
+		if bulkUpdateRes.Errors {
+			for _, resUpdateItem := range bulkUpdateRes.Items {
+				if resUpdateItem["update"].Status == 200 {
+					continue
+				} else {
+					log.Error(ctx, "error updating event to ES", err,
+						log.Data{
+							"response status": resUpdateItem["update"].Status,
+							"response.title:": resUpdateItem["update"].ID,
+						})
+					target--
+					continue
+				}
+			}
+		}
+	}
+	logData := log.Data{
+		"Events received":         len(events),
+		"Events inserted into ES": target,
+	}
+	log.Info(ctx, "ES updates", logData)
```

Comment:
I would combine line 139 to 144 into one message:

```suggestion
	log.Info(ctx, "documents bulk uploaded to elasticsearch", log.Data{
		"documents_received": len(events),
		"documents_inserted": target,
	})
```

The `log.Data` key value pairs must be structured like our responses from our APIs - so keys needed to replace whitespace with hyphens and be all lower cased :)